common/util.go

```diff
@@ -149,3 +150,18 @@ func GetConnectionAddr(ctx context.Context) string {
 	}
 	return from
 }
+
+// RatToFixed converts a big.Rat into a fixed point number represented as int64
+// according to supplied scale factor and maximum decimalplaces
+func RatToFixed(rat *big.Rat, scalingFactor int64, maxDecimalPlaces int) (int64, error) {
+	if rat == nil {
+		return 0, fmt.Errorf("reference to rat is nil")
+	}
+	scaled := new(big.Rat).Mul(rat, big.NewRat(scalingFactor, 1))
+	s := scaled.FloatString(maxDecimalPlaces)
+	intPart, err := strconv.ParseInt(strings.Split(s, ".")[0], 10, 64)
```

Comment:
Wow, that is painful, but I see why it's done this way.

Reply:
Could you elaborate on what exactly is painful 😆 .
The scaleFactor and decimals are arbitrary and we can still play with this but the workflow is what it is. sqlite-go doesn't have DECIMALS so we have to use int64.

Reply:
The intermediate string representation and parsing that is what's painful:
rat -> float -> string -> parse string -> int

Glancing through the docs again, could we do this instead:

ret, _ := new(big.Float).SetRat(scaled).Int64()

https://golang.org/pkg/math/big/#Float.SetRat

Reply:
Correct me if I'm wrong but `FloatString` despite its name is not actually a float conversion. 

Rather rat -> buffer -> string -> int64

Reply:
Sure, just that the stringification / parsing step could be removed.